### PR TITLE
[3.5] Backport e2e tests for livez/readyz.

### PR DESCRIPTION
**Pull request by siyuanfoundation**
Please read https://github.com/etcd-io/etcd/blob/main/CONTRIBUTING.md#contribution-flow.

Part of https://github.com/etcd-io/etcd/issues/16007

Followup PR of https://github.com/etcd-io/etcd/pull/17039

Original PR https://github.com/etcd-io/etcd/pull/16698

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -24,7 +24,7 @@ GO_BUILD_ENV=("CGO_ENABLED=0" "GO_BUILD_FLAGS=${GO_BUILD_FLAGS:-}" "GOOS=${GOOS}
 toggle_failpoints() {
   mode="$1"
   if command -v gofail >/dev/null 2>&1; then
-    run gofail "$mode" server/etcdserver/ server/mvcc/backend/ server/wal/
+    run gofail "$mode" server/etcdserver/ server/mvcc/ server/wal/ server/mvcc/backend/
```
Comment (serathius) [inline build.sh:27]: `make gofail-enable` and and `toggle_failpoints` in build.sh are separate clashing mechanisms. `toggle_failpoints` will always override `make gofail-enable` as it is run last. You need to either use one or the other.

Comment (siyuanfoundation) [reply build.sh:28]: removed `gofail-enable`

**Review by serathius — APPROVED** [review]

**Comment by serathius** [comment]
cc @ahrtr

File: Makefile [hunk 2]
```diff
@@ -565,3 +565,9 @@ pull-docker-functional:
 	$(info GO_VERSION: $(GO_VERSION))
 	$(info ETCD_VERSION: $(ETCD_VERSION))
 	docker pull gcr.io/etcd-development/etcd-functional:go$(GO_VERSION)
+
+# Failpoints
+GOFAIL_VERSION = $(shell cd tools/mod && go list -m -f {{.Version}} go.etcd.io/gofail)
+.PHONY: install-gofail
+install-gofail:
+	cd tools/mod; go install go.etcd.io/gofail@${GOFAIL_VERSION}
```
Comment (ahrtr) [inline Makefile:573]: Let's backport https://github.com/etcd-io/etcd/pull/17088 as well, in this PR or in a followup PR. Thanks.

**Review by ahrtr — APPROVED** [review]
LGTM with a minor comment.

Thanks

File: Makefile (hunk 2 above)
Comment (siyuanfoundation) [reply Makefile]: done

**Pull request merged**